### PR TITLE
README.md rewrite

### LINKED ISSUE #374: Incomplete System Requirements in README
I tried installing in a fresh Docker container and had to do two more steps before the `bundle install` succeeded.

Need some native C headers for `json` gem.

Ubuntu : `sudo apt-get install ruby-dev`
OSX : `xcode-select install`

 Need to have some postgresql headers for the `pg` gem.

Ubuntu : `sudo apt-get install libpq-dev`
OSX: `brew install postgresql`

### PR DESCRIPTION
**Pull request by jsyeo**
1. Add logo
2. Add link to slack channel
3. Remove production specific instructions
4. Add link to guide to setup rails

Fix #374

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,39 +1,52 @@
...
-    1. `bundle install --without ci:development:test` might be preferable for production installs.
-    2. `bundle install --without ci:production` might be preferable if you're only intending to
-       implement features.
- 2. `rake db:setup`. Run `RAILS_ENV=test rake db:setup` too if you are intending to run the test
-    suite. Run `RAILS_ENV=production rake db:setup` for a production server.
- 3. `RAILS_ENV=production rake assets:precompile` for production servers, or all the assets won't load.
- 4. `bin/rails server` to start the development app server. Production installs should configure
-    [Puma](http://puma.io) and be reverse-proxied.
- 5. You can log in with the default username and password:
+1. Ruby (Preferably >= 2.0.0)
```
Comment (lowjoel) [inline README.md]: Not preferably, we only test on 2.1 and 2.2. I do remember using some 2.1 features.

File: README.md [hunk 2]
```diff
@@ -1,39 +1,52 @@
...
+1. Ruby on Rails
```
Comment (lowjoel) [inline README.md]: I'm OCD, I number my list items in running order.

Comment (lowjoel) [inline README.md]: nodejs?

File: README.md [hunk 3]
```diff
@@ -1,39 +1,52 @@
...
+
+### Getting Started
+
+ 1. Download bundler to install dependencies
+
+    gem install bundler
+
+ 1. Install dependencies
+
+    bundle install --without ci:production
```
Comment (lowjoel) [inline README.md]: might want to prefix with `$`?

**Comment by lowjoel** [comment]
Are the problems mentioned in #374 mentioned somewhere in your new rewrite?

**Comment by jsyeo** [comment]
#374 problems can be fixed by following the [guide](https://gorails.com/setup/) I referenced.

**Comment by evansb** [comment]
I am not sure, I encountered the problem on a fresh Ubuntu system and the guide you mentioned is for OSX.

Anyway if someone wants to use Docker, I've written a working Dockerfile for Coursemology 2 that includes fixes for #374 and initial PostgreSQL setup.

https://github.com/evansb/coursemology-docker

**Comment by jsyeo** [comment]
The guide has instructions for Ubuntu.

**Comment by jsyeo** [comment]
https://gorails.com/setup/ubuntu/15.04

**Comment by jsyeo** [comment]
Anyway guiding people on how to set up rails is out of the scope of the
readme. It's not specific to our project.

**Comment by jsyeo** [comment]
On Jun 27, 2015 10:36 PM, "Evan Sebastian" <notifications@github.com> wrote:

> Anyway if someone wants to use Docker, I've written a working Dockerfile
for Coursemology 2 that includes fixes for #374 and initial PostgreSQL
setup.
>
> https://github.com/evansb/coursemology-docker

👍 can you put a link to your repo in the readme and send in a PR?

**Comment by lowjoel** [comment]
@evansb @jsyeo I'd rather that link be in the Wiki.

**Comment by lowjoel** [comment]
Uhh, did I fudge up the syntax for $? I thought it'd mark it preformatted code.

Also the username/password is merged on one line...

**Pull request merged**